app/controllers/subscriptions_auth_token_controller.rb

```diff
@@ -2,13 +2,39 @@ class SubscriptionsAuthTokenController < ApplicationController
   before_action :validate_params
 
   def auth_token
+    address = params.fetch(:address)
+    topic_id = params.fetch(:topic_id)
+    frequency = params.fetch(:frequency)
+
+    token = build_token(address: address, topic_id: topic_id, frequency: frequency)
+    email = build_email(address: address, token: token, topic_id: topic_id, frequency: frequency)
+
+    do_send email
     render json: {}, status: :ok
   end
 
+  def do_send(email)
+    DeliveryRequestWorker
+      .perform_async_in_queue(email.id, queue: :delivery_immediate_high)
+  end
+
   def validate_params
     ParamsValidator.new(expected_params).validate!
   end
 
+  def build_email(address:, token:, topic_id:, frequency:)
+    SubscriptionAuthEmailBuilder.call(address: address, token: token, topic_id: topic_id, frequency: frequency)
+  end
+
+  def build_token(address:, topic_id:, frequency:)
+    token_hash = {
+      address: address,
+      topic_id: topic_id,
+      frequency: frequency,
```

Comment:
We don't actually need the frequency in the token, but it doesn't do any harm to have it in there.